### ISSUE TITLE
Is there any way to protect premium video from Daydream screen recording?

### ISSUE THREAD

**Issue by maruyama-tomo** [googlevr/gvr-unity-sdk#883]
Summary:
I tried some protection related options but none of them worked as expected.
* Unity's Player Settings > Protect Graphics Memory checkbox => no effect
* Adding FLAG_SECURE window flag in a custom Activity as a native plugin => blackouts only thumbnail of my app in the Overview screen
In any case, Daydream screen recording (Home button + Volume up) was still active.
How can we protect our premium video contents from casual copy?

If a screen protection requires DRM, could you provide how to set missing License URL field?
Widevine could be an option since our company have CWIP.

Found using:
 * Google VR SDK version: 1.120.0, 1.130.1
 * Unity version: 2017.3.1p4, 2018.1.0b11
 * Phone manufacturer, model, and O/S version: ASUS_A002, Android 7.0
 * Viewer manufacturer & model: Daydream View

**Comment by rusmaxham** [comment]
I assume you have a Java plugin for managing your DRM video stream.  Also, if your video is only monoscopic 2D, then you can use Unity's [GVR Async Video Reprojection](https://docs.unity3d.com/Manual/VRDevices-GoogleVRVideoAsyncReprojection.html) to stream protected video in VR.

**Comment by maruyama-tomo** [comment]
@rusmaxham thank you for the information. but our videos are not only monoscopic, we would like to stream stereoscopic 180 and 360 VR videos. Are there any options?

My second question was partially solved. I could set License proxy URL by using [SetVideoPlayerSupportClassname](https://github.com/googlevr/gvr-unity-sdk/blob/v1.130.1/Assets/GoogleVR/Scripts/Video/GvrVideoPlayerTexture.cs#L823) to register a custom Java class that returns extended VideoPlayerFactory.
Then, I tested DRM video streaming. But I found the video is still screen-recordable **even if it is Widevine encrypted**.

**Comment by rusmaxham** [comment]
Is Daydream screen recording disabled if you check the Enable Video Surface and Use Protected Memory checkboxes under XR Settings > VR SDKs > Daydream?

**Comment by maruyama-tomo** [comment]
No, screen recording is enabled with those checkboxes on & Widevine encrypted mono 2D video.  ExoPlayer is GVR SDK's exoplayer-r1.5.11.aar, DASHAsyncRendererBuilder.drmCallback is replaced with my custom MediaDrmCallback using reflection.
Gvr Video Player Texture > Use Secure Path is unchecked. (If I check the Use Secure Path, it throws `DecoderInitializationException: Decoder init failed: OMX.qcom.video.decoder.avc.secure`)

**Comment by rusmaxham** [comment]
Hmmm..  the only way to protect video content at the moment is to use the GVR Async Video Reprojection facility in Unity, and due to its lack of support for non-quad projections, there isn't currently a way to prevent screen recording of it that I know of.

The secure path exception does seem like an issue.  Could you post a bugreport that captures the Secure Path exception for me to look at?

**Comment by maruyama-tomo** [comment]
[use-secure-path-error.log](https://github.com/googlevr/gvr-unity-sdk/files/1863147/use-secure-path-error.log)
The exception starts at line 375.

**Comment by maruyama-tomo** [comment]
@rusmaxham 
Is the Daydream screen recording detectable like an iOS's [`UIScreen.isCaptured`](https://developer.apple.com/documentation/uikit/uiscreen/2921651-iscaptured)?

**Comment by rusmaxham** [comment]
Unfortunately, not at this time.  I've filed a feature request with our platform team.
